aztk/spark/cli/cluster_create.py

```diff
@@ -69,19 +66,44 @@ def execute(args: typing.NamedTuple):
     log.info("-------------------------------------------")
 
     # create spark cluster
-    aztk.cluster.create_cluster(
-        cluster_conf.custom_scripts,
-        cluster_conf.uid,
-        cluster_conf.size,
-        cluster_conf.size_low_pri,
-        cluster_conf.vm_size,
-        cluster_conf.username,
-        cluster_conf.password,
-        cluster_conf.ssh_key,
-        cluster_conf.docker_repo,
-        cluster_conf.wait)
+    if cluster_conf.custom_scripts:
+        custom_scripts = [aztk_sdk.spark.models.CustomScript(name="", script=custom_script['script'], run_on=custom_script['runOn']) for custom_script in cluster_conf.custom_scripts]
+    else:
+        custom_scripts = None
+
+    cluster = aztk.client.create_cluster(
+        aztk_sdk.spark.models.ClusterConfiguration(
+            cluster_id=cluster_conf.uid,
+            vm_count=cluster_conf.size,
+            vm_low_pri_count=cluster_conf.size_low_pri,
+            vm_size=cluster_conf.vm_size,
+            custom_scripts=custom_scripts,
+            docker_repo=cluster_conf.docker_repo,
+            spark_configuration=aztk_sdk.spark.models.SparkConfiguration(
+                spark_defaults_conf=os.path.join(aztk_sdk.utils.constants.DEFAULT_SPARK_CONF_SOURCE, 'spark-defaults.conf'),
+                spark_env_sh=os.path.join(aztk_sdk.utils.constants.DEFAULT_SPARK_CONF_SOURCE, 'spark-env.sh'),
+                core_site_xml=os.path.join(aztk_sdk.utils.constants.DEFAULT_SPARK_CONF_SOURCE, 'core-site.xml'),
+                jars=[
+                    os.path.join(aztk_sdk.utils.constants.DEFAULT_SPARK_JARS_SOURCE, path) for path in os.listdir(aztk_sdk.utils.constants.DEFAULT_SPARK_JARS_SOURCE)
+                ]
+            )
+        ),
+        wait=cluster_conf.wait
+    )
+
+    if cluster_conf.username:
+        ssh_key = aztk.client.secrets_config.ssh_pub_key
+
```

Comment:
nitpick: single spaces within a function, multiple spaces between functions?

Reply:
I believe that is what I do here and throughout. Are you saying to do something different or am I missing something?

Reply:
For some reason I thought I saw two spaces around this if statement... probably my bad. Disregard.